### PR TITLE
RA-1696:Switch from 'success()' to 'done()' and 'error()' to 'fail()' in JQuery 3.4.1

### PR DESCRIPTION
**Pull request by reagan-meant**
Made the change because in JQuery 3.4.1 the ajax successful call back is done() after deprecation of success() and fail() with deprecation of error() as the error call back function
https://issues.openmrs.org/browse/RA-1696

### CODE REVIEW

File: omod/src/main/webapp/resources/scripts/custom/visits.js [hunk 1]
```diff
@@ -131,7 +131,7 @@ visit.showEditVisitDateDialog = function(visitId) {
             actions: {
                 confirm: function() {
                     var url = emr.fragmentActionLink("coreapps", "visit/visitDates", "setDuration");
-                    $.getJSON(url, $('#edit-visit-dates-dialog-form-' + visitId).serialize()).success(function(data) {
+                    $.getJSON(url, $('#edit-visit-dates-dialog-form-' + visitId).serialize()).done(function(data) {
                             if (data.success) {
```
Comment (mozzy11) [inline omod/src/main/webapp/resources/scripts/custom/visits.js]: Thanks @reagan-meant  .Is this the only occuranceof    .succes()  ??

Comment (reagan-meant) [reply omod/src/main/webapp/resources/scripts/custom/visits.js]: @mozzy11 I have changed the error method to fail

File: omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js [hunk 2]
```diff
@@ -52,9 +52,9 @@ $(function() {
 	    		if(encounterDetailsSection.html() == "") { encounterDetailsSection.html("<i class=\"icon-spinner icon-spin icon-2x pull-left\"></i>");}
 	    jq.getJSON(
 	        		emr.fragmentActionLink("htmlformentryui", "htmlform/viewEncounterWithHtmlForm", "getAsHtml", { encounterId: id })
-	        ).success(function(data){
+	        ).done(function(data){
 	            encounterDetailsSection.html(data.html);
-	        }).error(function(err){
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js]: Is this change related to what this ticket is all about?

File: omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js [hunk 3]
```diff
@@ -68,9 +68,9 @@ $(function() {
 	    		if(encounterDetailsSection.html() == "") { encounterDetailsSection.html("<i class=\"icon-spinner icon-spin icon-2x pull-left\"></i>");}
 	        jq.getJSON(
 	            emr.fragmentActionLink("coreapps", "visit/visitDetails", "getEncounterDetails", { encounterId: id })
-	        ).success(function(data){
+	        ).done(function(data){
 	            encounterDetailsSection.html(displayTemplate(data));
-	        }).error(function(err){
+	        }).fail(function(err){
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js]: Is this change related to what this ticket is all about?

File: omod/src/main/webapp/resources/scripts/fragments/visitDetails.js [hunk 4]
```diff
@@ -63,7 +63,7 @@ function loadTemplates (visitId, patientId, fromEncounter, encounterCount, curre
                     });
                 }
                 
-            }).error(function(err) {
+            }).fail(function(err) {
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/scripts/fragments/visitDetails.js]: Is this change related to what this ticket is all about?

File: omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js (hunk 2 above)
Comment (reagan-meant) [reply omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js]: The deprecation of success() to done() came together with the deprecation of error () to fail().....so it seemed logical to do both in the same sitting....What would be your suggestion?....Should I modified the ticket description to accommodate both or I shud have a different ticket for that too or I ignore it all together?

File: omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js (hunk 3 above)
Comment (reagan-meant) [reply omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js]: The deprecation of success() to done() came together with the deprecation of error () to fail().....so it seemed logical to do both in the same sitting....What would be your suggestion?....Should I modified the ticket description to accommodate both or I shud have a different ticket for that too or I ignore it all together?

File: omod/src/main/webapp/resources/scripts/fragments/visitDetails.js (hunk 4 above)
Comment (reagan-meant) [reply omod/src/main/webapp/resources/scripts/fragments/visitDetails.js]: The deprecation of success() to done() came together with the deprecation of error () to fail().....so it seemed logical to do both in the same sitting....What would be your suggestion?....Should I modified the ticket description to accommodate both or I shud have a different ticket for that too or I ignore it all together?

File: omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js (hunk 2 above)
Comment (dkayiwa) [reply omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js]: If you want both, then you would need to update both the ticket description and commit message to reflect what you are doing

Comment (reagan-meant) [reply omod/src/main/webapp/resources/scripts/fragments/patientdashboard/encountertemplate/defaultEncounterTemplate.js]: Let me modify then... Thanks

File: omod/src/main/webapp/resources/scripts/fragments/visitDetails.js (hunk 4 above)
Comment (reagan-meant) [reply omod/src/main/webapp/resources/scripts/fragments/visitDetails.js]: @Ruhanga do you mean this?

Comment (Ruhanga) [reply omod/src/main/webapp/resources/scripts/fragments/visitDetails.js]: Ahh, yes. Nice catch. Thanks I hadn't see this change.

**Comment by dkayiwa** [comment]
@reagan-meant did you see the above comment?

**Comment by reagan-meant** [comment]
I responded to it in the forth portion of this thread....I had fixed it aswell

**Comment by mozzy11** [comment]
Thanks @reagan-meant  and @Ruhanga
